### DISCUSSION TITLE
Error while using Uvicorn main:app --reload

### DISCUSSION THREAD

**Discussion by KarthikChimkode** [fastapi/fastapi#11435] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

app = FastAPI()


@app.get("/")
def root():
    return {"message": "welcome to my api"}

@app.get("/post")
def get_posts():
    return {"data": "this is tour data"}
```

### Description

i am changing the path in @app.get then it works fine with univorn main:app but if i use --reload it throws 
INFO:     Shutting down
INFO:     Waiting for application shutdown.
INFO:     Application shutdown complete.
INFO:     Finished server process [14500]
Process SpawnProcess-1:
Traceback (most recent call last):
  File "C:\Users\karth\AppData\Local\Programs\Python\Python311\Lib\asyncio\runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\karth\AppData\Local\Programs\Python\Python311\Lib\asyncio\base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
asyncio.exceptions.CancelledError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\karth\AppData\Local\Programs\Python\Python311\Lib\multiprocessing\process.py", line 314, in _bootstrap
    self.run()
  File "C:\Users\karth\AppData\Local\Programs\Python\Python311\Lib\multiprocessing\process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\karth\Projects\Fastapi\venv\Lib\site-packages\uvicorn\_subprocess.py", line 78, in subprocess_started
    target(sockets=sockets)
  File "C:\Users\karth\Projects\Fastapi\venv\Lib\site-packages\uvicorn\server.py", line 65, in run
    return asyncio.run(self.serve(sockets=sockets))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\karth\AppData\Local\Programs\Python\Python311\Lib\asyncio\runners.py", line 190, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "C:\Users\karth\AppData\Local\Programs\Python\Python311\Lib\asyncio\runners.py", line 123, in run
    raise KeyboardInterrupt()
KeyboardInterrupt
INFO:     Started server process [16988]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
this

### Operating System

Windows

### FastAPI Version

0.110.1

### Pydantic Version

2.6.4

### Python Version

Python 3.11.3

**Comment by angad-singhh** [comment]
I am getting the same error, if found any solution comment down

**Comment by KarthikChimkode** [comment]
No not found any solution use it without -- reload that's the only way it's
not throwing errors

**Comment by angad-singhh** [comment]
I checked the uvicorn GitHub issues for this, found the fix : resolve, the problem only exist in the current version of uvicorn==0.29.0 , rest it is working fine in any previous versions like 0.28.1 or 0.27.1 etc.

**Comment by KarthikChimkode** [comment]
Awesome, thanks for the update!  And how can I change the version

**Reply by angad-singhh** [reply]
You can download any FastAPI version using pip install fastapi=={version name}

**Comment by KarthikChimkode** [comment]
Thank you!!!

**Comment by YuriiMotov** [comment]
I guess this was fixed somewhere. Woks fine with current versions
